src/auth/auth-config.ts

```diff
@@ -806,6 +880,17 @@ export class OIDCConfig implements OIDCAuthProviderConfig {
     // When enabled is undefined, it takes its default value of false.
     this.enabled = !!response.enabled;
     this.displayName = response.displayName;
+
+    if (typeof response.clientSecret !== 'undefined') {
+      this.clientSecret = response.clientSecret;
+    }
+    // If we do not set responseType, we have idToken flow
```

Comment:
we don't need to do this - the backend will default to the ID token flow

Reply:
Yes, I think line 892 is the case setting to ID token flow by default, this section I am fetching the secret(for need if this is code flow).

Reply:
The server response should already have a responseType set to ID token so we wouldn't have to set it here IIRC

Reply:
Thanks for explaining. I experimented what you said in integration test, by calling createProviderConfig or getProviderConfig triggering RESTAPI with OIDCConfig(if doesnt include responseType, the config I get doesn't include responseType as well).
Therefore, I think it is necessary to include the logic (auto-set idToken flow)

Reply:
The backend definitely returns this - it must be not getting set somewhere else. Here's a sample response when I make the call myself:

{
  "name": "projects/projectId/oauthIdpConfigs/oidc.testing-1234",
  "clientId": "CLIENT_ID",
  "issuer": "https://google.com",
  "displayName": "displayName",
  "responseType": {
    "idToken": true
  }
}

Might be worth looking at the backend response in your integration test, and see why it's not part of the response here

Reply:
I thought the original integration test covered the case you mentioned, and it didn't return an error. Thanks for correction.